pageserver/src/remote_storage/rust_s3.rs

```diff
@@ -23,15 +25,38 @@ impl S3ObjectKey {
         &self.0
     }
 
-    fn download_destination(&self, pageserver_workdir: &Path) -> PathBuf {
-        pageserver_workdir.join(self.0.split(S3_FILE_SEPARATOR).collect::<PathBuf>())
+    fn download_destination(
+        &self,
+        pageserver_workdir: &Path,
+        prefix_to_strip: Option<&str>,
+    ) -> PathBuf {
+        let path_without_prefix = match prefix_to_strip {
+            Some(prefix) => self
+                .0
+                .strip_prefix(prefix)
+                .with_context(|| {
+                    format!(
+                        "Could not strip prefix '{}' from S3 object key '{}'",
+                        prefix, self.0
+                    )
+                })
+                .unwrap(),
+            None => &self.0,
+        };
+
+        pageserver_workdir.join(
+            path_without_prefix
+                .split(S3_FILE_SEPARATOR)
```

Comment:
My reasoning behind that was to ensure that the path separators are placed by the library code right, also hoped that some kind of validation and normalization is done (empty path segments thrown off to eliminate `//` cases, etc.).

Since we're not targeting Windows, do you think we should omit that?